journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -546,6 +539,33 @@ private FileChannel openChannel(final File file) {
     return segments.values();
   }
 
+  private JournalSegmentDescriptor readDescriptor(final File file) {
+    final int descriptorLength = JournalSegmentDescriptor.getEncodingLength();
+    if (file.length() < descriptorLength) {
```

Comment:
Can this break backward compatability? Assume we added a new field to the segment descriptor and the broker is with the new version. Then `JournalSegmentDescriptor.getEncodingLength()` will include the new field. If it is reading a segment which was written with the old version, then it is possible that file.length = old descriptor length which is smaller, right?

Reply:
When we have multiple versions we'll use the magic byte to know which version we're dealing with and what format to expect. Currently there's only one possible value so it's fine. I expect that to change next week when I work on broadening the corruption detection in the descriptor

Reply:
If we are only adding new fields to the SBE message, we don't have to change to the version number in magic byte.

Reply:
What would happen if we don't check this condition here? Would we be able to detect that the descriptor is partial when reading the sbe message?

Reply:
We can add a minimum length (i.e., what it is now) method to the descriptor and check against that. We allocate a buffer using the encoding size since it might have more but we only assume it's corruption if the length is smaller than the minimum. If we change the format then that might decrease of course but if we keep the SBE format then it should be the same since we can't remove fields